core/test/matrix/sparsity.cpp

```diff
@@ -0,0 +1,212 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2019, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/matrix/sparsity.hpp>
+
+
+#include <memory>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/base/dim.hpp>
+
+
+namespace {
+
+
+class Sparsity : public ::testing::Test {
+protected:
+    using Mtx = gko::matrix::Sparsity<>;
+
+    Sparsity()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(gko::matrix::Sparsity<>::create(exec, gko::dim<2>{2, 3}, 4))
+    {
+        Mtx::index_type *c = mtx->get_col_idxs();
+        Mtx::index_type *r = mtx->get_row_ptrs();
+        r[0] = 0;
+        r[1] = 3;
+        r[2] = 4;
+        c[0] = 0;
+        c[1] = 1;
+        c[2] = 2;
+        c[3] = 1;
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::unique_ptr<Mtx> mtx;
+
+    void assert_equal_to_original_mtx(const Mtx *m)
+    {
+        auto c = m->get_const_col_idxs();
+        auto r = m->get_const_row_ptrs();
+        ASSERT_EQ(m->get_size(), gko::dim<2>(2, 3));
```

Comment:
You are missing the following tests:  `get_col_idxs()`, `get_row_ptrs()`, `get_value()` and `get_const_value()`.
I think for testing `get_col_idxs()`, `get_row_ptrs()` and `get_value()`, you can simply test against the pointer of their const counterpart:
```
ASSERT_EQ(m->get_const_col_idxs(), m->get_col_idxs());
ASSERT_EQ(m->get_const_row_ptrs(), m->get_row_ptrs());
ASSERT_EQ(m->get_const_value(), m->get_value());
```
Then, you are only missing the test `EXPECT_EQ(m->get_const_value()[0], 1.0);`

Reply:
Actually, I think it would be better if you set the value in the beginning to something different than `1.0` (simply `2.0` should be fine), so you can check here if it will also be changed, or not.